src/Microsoft.Azure.ServiceBus/SessionClient.cs

```diff
@@ -236,9 +245,18 @@ await this.RetryPolicy.RunOperation(() => session.GetSessionReceiverLinkAsync(se
                     this.EntityPath,
                     exception);
 
+                if (isDiagnosticsEnabled)
+                {
+                    this.diagnosticSource.ReportException(exception);
+                }
+
                 await session.CloseAsync().ConfigureAwait(false);
                 throw AmqpExceptionHelper.GetClientException(exception);
             }
+            finally
+            {
+                this.diagnosticSource.AcceptMessageSessionStop(activity, sessionId, serverWaitTime, session, acceptMessageSessionTask?.Status);
```

Comment:
>session [](start = 100, length = 7)

we don't need to pass the `session` object. Don't think its very useful. 
If you are removing this, make sure instead of passing `sessionId`, pass `session.SessionId`